src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php

```diff
@@ -165,19 +226,77 @@ public function configureOptions(OptionsResolver $resolver)
             return $em;
         };
 
+        // deprecation note
+        $propertyNormalizer = function (Options $options, $propertyName) {
+            trigger_error('The "property" option is deprecated since version 2.7 and will be removed in 3.0. Use "choice_label" instead.', E_USER_DEPRECATED);
```

Comment:
This normalizer is called for BC by default in the choice_label option. So this will trigger the deprecation notice even if not used. I guess we have to test if `null !== $propertyName` and only then trigger the notice in order to only warn when the option is actually used.